pkg/apis/componentconfig/validation/validation.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/apis/componentconfig"
+	containermanager "k8s.io/kubernetes/pkg/kubelet/cm"
+)
+
+// MaxCrashLoopThreshold is the maximum allowed KubeletConfiguraiton.CrashLoopThreshold
+const MaxCrashLoopThreshold = 10
+
+// ValidateKubeletConfiguration validates `kc` and returns an error if it is invalid
```

Comment:
What if we just wrote validation helper functions for each sub-component in this file instead?

I'd really like the validation of the config type to (a) live close to where the config type is defined and (b) run all the validation very early in the Kubelet's lifecycle. 

If we had each component expose its own validation function, I think I'd also want each component to define it's own subset of the config type close to that validation function. I worry this will be confusing, as it will spread the API definition and validation around and could encourage too much granularity in the API.

Basically I want to find the Kubelet's entire API in one place, and also find the valid ranges for those parameters in one place. And I want it to require as few parameters as possible.